api/pyproject.toml

```diff
@@ -6,7 +6,7 @@ packages = [{ include = "src" }]
 authors = ["Nava Engineering <engineering@navapbc.com>"]
 
 [tool.poetry.dependencies]
-python = "^3.10"
+python = "^3.12"
```

Comment:
Technically 3.12 didn't add much in terms of features (a lot of under the hood things from the look of things) but figured we should say it needs to be more up to date. 3.11 is definitely the minimum though as there are features we use that were added in that